cpp/src/barretenberg/honk/composer/composer_helper/turbo_plonk_composer_helper.cpp

```diff
@@ -0,0 +1,187 @@
+#include "turbo_plonk_composer_helper.hpp"
+#include "barretenberg/honk/circuit_constructors/turbo_circuit_constructor.hpp"
+#include "barretenberg/ecc/curves/bn254/scalar_multiplication/scalar_multiplication.hpp"
+#include "barretenberg/numeric/bitop/get_msb.hpp"
+#include "barretenberg/plonk/proof_system/widgets/random_widgets/permutation_widget.hpp"
+#include "barretenberg/plonk/proof_system/widgets/transition_widgets/turbo_arithmetic_widget.hpp"
+#include "barretenberg/plonk/proof_system/widgets/transition_widgets/fixed_base_widget.hpp"
+#include "barretenberg/plonk/proof_system/widgets/transition_widgets/turbo_logic_widget.hpp"
+#include "barretenberg/plonk/proof_system/widgets/transition_widgets/turbo_range_widget.hpp"
+#include "barretenberg/plonk/proof_system/commitment_scheme/kate_commitment_scheme.hpp"
+#include "barretenberg/plonk/proof_system/widgets/transition_widgets/transition_widget.hpp"
+#include "barretenberg/plonk/proof_system/widgets/transition_widgets/turbo_arithmetic_widget.hpp"
+#include "permutation_helper.hpp"
+using namespace barretenberg;
+
+namespace bonk {
+
+/**
+ * Compute proving key
+ *
+ * 1. Load crs.
+ * 2. Initialize this.circuit_proving_key.
+ * 3. Create constraint selector polynomials from each of this composer's `selectors` vectors and add them to the
+ * proving key.
+ * 4. Compute sigma polynomial
+ *
+ * @return Pointer to the initialized proving key updated with selector polynomials.
+ * */
+template <typename CircuitConstructor>
+std::shared_ptr<bonk::proving_key> TurboPlonkComposerHelper<CircuitConstructor>::compute_proving_key(
+    const CircuitConstructor& circuit_constructor)
+{
+    if (circuit_proving_key) {
+        return circuit_proving_key;
+    }
+    const size_t minimum_circuit_size = 0;
+    const size_t num_randomized_gates = NUM_RANDOMIZED_GATES;
+    // Initialize circuit_proving_key
+    // TODO(#229)(Kesha): replace composer types.
+    circuit_proving_key = initialize_proving_key(circuit_constructor,
+                                                 crs_factory_.get(),
+                                                 minimum_circuit_size,
+                                                 num_randomized_gates,
+                                                 plonk::ComposerType::TURBO);
+    // Compute lagrange selectors
+    construct_lagrange_selector_forms(circuit_constructor, circuit_proving_key.get());
+    // Make all selectors nonzero
+    enforce_nonzero_polynomial_selectors(circuit_constructor, circuit_proving_key.get());
+
+    // Compute selectors in monomial form
```

Comment:
Comment doesn't mention coset fft -- but also the function names are so descriptive you could probably just do away with the comments altogether.